homeassistant/components/pglab/config_flow.py

```diff
@@ -0,0 +1,56 @@
+"""Config flow for PG LAB Electronics integration."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.helpers.service_info.mqtt import MqttServiceInfo
+
+from .const import DISCOVERY_TOPIC, DOMAIN
+
+CONF_DISCOVERY_PREFIX = "discovery_prefix"
+
+
+class PGLabFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    VERSION = 1
+
+    async def async_step_mqtt(
+        self, discovery_info: MqttServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle a flow initialized by MQTT discovery."""
+
+        await self.async_set_unique_id(DOMAIN)
+
+        # Validate the message, abort if it fails.
+        if not discovery_info.topic.endswith("/config"):
+            # Not a PGLab Electronics discovery message.
+            return self.async_abort(reason="invalid_discovery_info")
+        if not discovery_info.payload:
+            # Empty payload, Unexpected payload.
+            return self.async_abort(reason="invalid_discovery_info")
+
+        return await self.async_step_confirm()
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a flow initialized by the user."""
+        return self.async_abort(reason="not_supported")
```

Comment:
I don't agree with Joost, I think we should allow this.

Reply:
I discussed with joost and a few others on Discord, and the conclusion is there should be a user flow, which should:
- Check the MQTT integration is configured and connected to the broker, and error if it isn't. You can call `mqtt.is_connected` to check this, it will raise a `KeyError` if there's no MQTT configuration, and `False` if `mqtt` is not connected to the broker.
- Show a message explaining the devices need to be connected to the same broker as the Home Assistant mqtt integration's client for them to be found

Reply:
I agree the possibility for the user to add manually the integration. I'm going to show two different messages when the integration is added with auto discovered and when it is added manually.   In the second case the message is as suggested in your last point.